test/app/lib/cinematic/commandRunner.spec.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env jasmine */
+import CommandRunner from '../../../../app/views/play/cinematic/Command/CommandRunner'
+import AbstractCommand from '../../../../app/views/play/cinematic/Command/AbstractCommand'
+import Promise from 'bluebird'
+const utils = require('../../utils.coffee')
+
+Promise.config({
+  cancellation: true
+})
+
+const sleep = ms => new Promise((resolve, reject) => setTimeout(resolve, ms))
+
+// Command that we use to test commandRunner functionality.
+class SpyCommand extends AbstractCommand {
+  constructor (spyRunner, ms, spyCancel) {
+    super()
+    this.run = () => sleep(ms).then(spyRunner)
+
+    // We can also spy on cancel calls.
+    if (typeof spyCancel === 'function') {
+      this.cancel = promise => {
+        promise.cancel()
+        spyCancel()
+        return Promise.resolve()
+      }
+    }
+  }
+}
+
+describe('commandRunner', () => {
+  it('runs all commands provided', utils.wrapJasmine(async () => {
+    const spy = jasmine.createSpy('commandPromiseSpy')
+    const cancelSpy = jasmine.createSpy('cancelSpy')
+
+    const commands = []
+    const commandLength = 5 + Math.floor((Math.random() * 10))
+    for (let i = 0; i < commandLength; i++) {
+      commands.push(new SpyCommand(spy, 100, cancelSpy))
+    }
+    const c = new CommandRunner(commands)
+    await c.run()
+
+    expect(spy).toHaveBeenCalledTimes(commandLength)
+    expect(cancelSpy).not.toHaveBeenCalled()
+  }))
+
+  it('cancels commands immediately correctly', utils.wrapJasmine(async () => {
+    const spy = jasmine.createSpy('commandPromiseSpy')
+    const cancelSpy = jasmine.createSpy('cancelSpy')
+
+    const commands = []
+    const commandLength = 1 + Math.floor((Math.random() * 10))
```

Comment:
I like that we test many permutations here, but it also brings up a question of randomness in our tests. What if a number (let's say 1) is the only permutation with an issue. What if that permutation never runs, randomness being what it is? Could we reliably say we are testing for all the different cases here? 

Having gone back and forth on this a bit, I now wonder if testing the known bounds on purpose is more reliable and useful than a test with randomness in it.